htseq2multx.pl

```diff
@@ -459,15 +467,25 @@ sub runFastqMultx
           }
       }
 
-    if($CHILD_ERROR || $make_error_fatal)
+    waitpid($pid, 0);
+    my $exit_code = $CHILD_ERROR >> 8;
```

Comment:
NB: Recommend making this a "constant" using the [Readonly](https://metacpan.org/pod/Readonly) or [Const::Fast](https://metacpan.org/pod/Const::Fast) module or the ["constant" pragma](https://perldoc.perl.org/constant) instead of the hardcoded "`8`".

Reply:
Bit shifting by 8 to the the exit code of a finished child process is as far as I know, a universal perl standard that has not changed since I've been coding perl. In fact `$CHILD_ERROR` could otherwise be `-1` in normal circumstances (if I'd done the fork in this script), but `IO::Pipe::Producer catches this case before exiting and exits with a value greater than or equal to `-1`, though it is possible there could exist an edge case that I'm not aware of, e.g. a dependent package rudely exiting for me.

I suppose I will add a constant, but I cannot envision it ever effectively being useful.

- [ ] Create a constant to hold the bit shift amount.

Reply:
Ah, that makes sense, no need to use a constant here then. Been a while since I dealt with this exit codes in Perl.